### PR TITLE
Support custom font path with typst

### LINKED ISSUE #7585: Support setting custom fonts with typst
From https://github.com/quarto-dev/quarto-cli/discussions/7580

Typst allows to use `--font-path` or `TYPST_FONT_PATHS` env var to set custom fonts folder in the search path
https://typst.app/docs/reference/text/text/#parameters-font

However, it does seem that both can't be use at the same time, e.g. doing 
````
# with Agbalumo custom font
$ tree
.
├── myfonts
│   └── Agbalumo-Regular.ttf
├── test.pdf
└── test.typ

# Found
$ TYPST_FONT_PATHS=./myfonts quarto typst fonts
Agbalumo
Alegreya
<truncated>

# Not found
$ TYPST_FONT_PATHS=./myfonts quarto typst fonts --font-path ./dummy
Alegreya
Alegreya SC
<truncated>
````

This is confirm looking at Rust code where the environment variable is a fallback
https://github.com/michidk/typst/blob/1235d52c52998e3fde6a34db8b18cb448a1e73a6/cli/src/args.rs#L9-L11

And in Quarto we are setting `--font-path`  since https://github.com/quarto-dev/quarto-cli/commit/cb41de7e3e1506e697b3362c42735b4781a276c9 to support for nicer callout output
https://github.com/quarto-dev/quarto-cli/blob/eb36ef9cf030b6515672df9090dc9e275b50c505/src/core/typst.ts#L31-L38

So it seems we need to adjust this usage to allow for custom font directory. 

## Solutions 

* Allowing a new `font-path` YAML options for typst to be pass as CLI flag 
````
$ quarto typst fonts --font-path ./myfonts --font-path ./dummy
Agbalumo
Alegreya
<truncated>
````

* Checking for `TYPST_FONT_PATHS` and pass it as a CLI flag

* Not using `--font-path` internally for our usage, but appending our internal font path to `TYPST_FONT_PATHS` which support some directories (comma separated I believe) 
	* which does not seem possibly for now... https://github.com/typst/typst/issues/1708

### PR DESCRIPTION
**Pull request by cderv**
Closes #7585 

If `TYPST_FONT_PATHS` is set, pass it through, otherwise our internal use of `--font-path` will override the value

Next step would be to support `font-path` yaml option probably with same rule as in typst: 
* If `font-path` is set, then `TYPST_FONT_PATHS` is ignored. 

Or we could do our own rule and also append...

### CODE REVIEW

**Comment by cderv** [comment]
I did add support for something like 
````
format: 
  typst:
      mainfont: "Agbalumo"
      font-paths: myfonts
````

where `Agbalumo` font lives locally  in `myfonts` folder

File: tests/docs/smoke-all/typst/custom-fonts.qmd [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+title: font typst
+format: 
+  typst:
+      mainfont: "Agbalumo"
+      font-paths: myfonts
+#TODO: How to test the PDF for correct font usage ?
```
Comment (cscheid) [inline tests/docs/smoke-all/typst/custom-fonts.qmd:7]: Just FWIW, I don't know either :(

I think we should spend some time in 1.5 figuring out how to test the content of PDFs more explicitly. That would be useful for all of our PDF output generation.

**Review by cscheid — APPROVED** [review]
This looks great, thank you!

**Comment by cderv** [comment]
Ok great; i'll add to the typst doc and then merge.

**Pull request merged**